### ISSUE TITLE
Écriture des changements de preset défectueuse dans DP (ancien: Preset 0)

### ISSUE THREAD

**Issue by Normandeau** [GRIS-UdeM/ControlGRIS#128]
V114
Il existe toujours un preset 0 dans ControlGRIS. Pourquoi? Cela porte à confusion. Si on ne porte pas attention et qu'on met la valeur au minimum, on peut très bien penser qu'on met la valeur à 1, alors qu'en fait on la met à 0 (tous les autres plugiciels commencent à 1). Si ce preset 0 est encore utile, alors, il faudra l'indiquer dans le manuel. Mais disons qu'en comparaison des autres plugiciels, c'est une incongruité. Il faudrait que je fouille dans les anciennes issues, car je me souviens qu'on a déjà discuté de cela.

Logic
On peut pas enregistrer les Presets en mode d'écriture d'automation. Seule l'écriture manuelle fonctionne. Le rappel des automations de presets fonctionne cependant.

**Comment by SamuelBeland** [comment]
Tu as raison, le preset 0 n'est plus utile depuis 1.1.2.

Je suis curieux par contre de la réaction des DAW à ce changement quand viendra le temps de charger des automations enregistrées dans l'ancien range.

Je travail sur le changement.

**Comment by SamuelBeland** [comment]
_Après avoir discuté avec @Normandeau, j'écris un petit résumé ici à des fins d'archives_ :

Dans mes tests avec Reaper, les valeurs d'automations semblent converties à l'interne dans le range [ 0 - 1 ] lors de la sauvegarde des projets. En changeant le range depuis [ 0 - 50 ] vers [ 1 - 50 ], ça aurait l'effet de pousser vers le haut les valeurs d'automation du champs _preset_ de la façon suivante : 

0 = 1
1 = 1.98
2 = 2.96
[ ... ]
48 = 48.04
49 = 49.02
50 = 50

Évidement, c'est un problème qui ne toucherait que les projets enregistrés dans une version précédente.

Ça serait donc un breaking change.

**Comment by Normandeau** [comment]
Oui, c'est trop risqué de faire le changement. On va laisser cela tel quel, et je mettrai une note dans le prochain manuel.

**Comment by Normandeau** [comment]
V119
Il y a un preset 0 disponible dans ControlGRIS pour éviter le conflit entre deux informations contradictoires au sujet de l'automation. En effet, si on enregistre l'automation des numéros des presets, et qu'ensuite on enregistre de l'automation de position (X, Y), ces dernières remplaçaient les Presets déjà enregistrés par un Preset 0:

![Capture d’écran, le 2021-07-06 à 16 14 23](https://user-images.githubusercontent.com/8959779/124661964-331d5680-de76-11eb-9158-d9bc3558e072.jpg)

Or depuis au moins la version 1.12, ce n'est plus le cas, ce qui fait les automations sont assez chaotiques:

![Capture d’écran, le 2021-07-06 à 16 09 58](https://user-images.githubusercontent.com/8959779/124662162-74156b00-de76-11eb-949e-0b4f5ade158c.jpg)

 Pourquoi a-t-on retirer cette fonctionnalité, qui est très bien décrite dans le manuel? C'est pourquoi je viens de la vérifier:
«It's possible to record the Presets’ number in the sequencer. But to avoid contradictory information between automated trajectories and automated presets, the latter is replaced by a 0 value (no preset) during the recording of a trajectory.»

**Comment by Normandeau** [comment]
V1.20
La solution ici est de faire en sorte que dès qu'un preset est écrit en mode d'écriture de l'automation, les coordonnées x et y enregistrées dans celui-ci soient aussi écrits sur les courbes d'automation. De telle sorte que la position de la source du preset soit parfaitement cohérente avec celle qui est écrit dans l'automation. De cette manière, il n'y aura pas de conflit entre les deux et par conséquent pas de glitch au moment du passage d'un nouveau preset. Et on oublie le preset 0 (qui reste quand même là pour ne pas changer l'échelle des presets. Je vais mettre une note dans le manuel pour indiquer que c'est une relique...)

**Comment by Normandeau** [comment]
V121
Ça fonctionne comme prévu dans Reaper et Logic, en mode Latch:

https://user-images.githubusercontent.com/8959779/126875193-2341a820-2bdc-41ce-a7dd-9b7a926eebf7.mov

Mais partiellement dans DP (10 ou 11, c'est pareil), en mode Latch ou Overwrite (je peux toujours corriger à la main, mais ce n'est pas vraiment pratique...). On voit que les points de changement sont écrits temporairement jusqu'au suivant par lequel il est effacé. En lecture, c'est très smooth...

https://user-images.githubusercontent.com/8959779/126875473-29221794-bddd-4250-a24f-a9f3c5c0d184.mov

Et pratiquement pas en mode Touch (ici on retrouve la contradiction qu'on essayait d'annuler entre les courbes X et y et le preset automatisé):

https://user-images.githubusercontent.com/8959779/126875264-7bb33288-5bd2-49d5-9882-a116b234005c.mov

Je vais voir s'il y a moyen de corriger cela dans DP.

**Comment by NicolaGiannini** [comment]
Je confirme que dans Logic ça fonctionne comme prévu.

**Comment by Normandeau** [comment]
V122
Toujours le même problème dans DP. On verra cela après la version 3. On a déjà eu un problème comme cela. Je vais chercher.
Voilà à quoi ça ressemble en lecture:

https://user-images.githubusercontent.com/8959779/127404696-d9d34c72-7fdd-4050-8ff5-1e3b25501470.mov

**Comment by Normandeau** [comment]
V145
Dans DP, les changements d'un preset à l'autre ne sont pas des points discrets. Comme décrit jusqu'ici, chaque fois qu'on change de preset, DP fait un courbe qui passe progressivement d'un point à l'autre (voir les courbes de V122. Ça ne devrait pas être le cas.

**Comment by Normandeau** [comment]
V200RC2
Ce comportement n'est toujours pas réglé dans DP. Cela a fiat l'objet d'une déclaration sur le site de MOTU depuis le 23 septembre... 2021! Avec un rappel le 18 octobre 2024...
Cela étant dit, peu importe la situation dans DP, cela ne règle pas le conflit qui existe entre la position des sources lorsqu'elle est appelée par un preset et celle qui est appelée par l'écriture directe. C'est à cela que servait le preset 0.
À valider dans les autres DAWs.

**Comment by Normandeau** [comment]
Reaper 7 et Logic 11
Il semble y avoir une hiérarchie ici: les courbes enregistrés des paramètres X et Y ont préséance sur les presets. Ceux-ci changent mais ne commandent pas les positions X et Y.

Live 11 et DP11
Ici c'est différent, les valeurs entrent en conflit. Les courbes enregistrées sont lues mais dès qu'un preset est lu à son tour, la source fait un saut pour revenir ensuite à la position des courbes écrites.

**Comment by Normandeau** [comment]
V200
Problème relié à DP.  Signalé à MOTU en 2021. Aucun changement. Conseil: Passer à un autre SAN!
